rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -722,16 +722,15 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 			// closest to us
 			const CGroundBlockingObjectMap::BlockingMapCell& cell = groundBlockingObjectMap.GetCellUnsafeConst(buildInfo.pos);
 
-			const CSolidObject* o = nullptr;
 			const CUnit* u = nullptr;
 
 			// look for any blocking assistable buildee at build.pos
 			for (size_t i = 0, n = cell.size(); i < n; i++) {
-				const CUnit* cu = dynamic_cast<const CUnit*>(o = cell[i]);
+				const CUnit* cu = dynamic_cast<const CUnit*>(cell[i]);
 
 				if (cu == nullptr)
 					continue;
-				if (allyteam != cu->allyteam)
+				if (allyteam != cu->allyteam) // NB: not teamHandler.Ally because allies don't share LoS
```

Comment:
I don't think LoS is relevant here.
The question is whether the default action for trying to build a building on the same spot an ally is building should be to assist.

Reply:
I think whether to default to assisting for allies is a valid but orthogonal question because the new stacking bug was about starting construction on top of finished buildings (which are unassistable). I now agree LoS isn't relevant, the real reason `teamHandler.Ally` can't be used is that `GetClosestFriendlyUnit` (which alongside the subsequent `buildInfo.FootPrintOverlap` are the fallback preventing stacking on finished, ie. unassistable buildings) works only on the allyteam and not its allies. Personally I don't think allies should be assisted by default (since they are really just temporarily ceasefired enemies) but I am probably not the one to decide here and I mostly just wanted to spend 5 seconds writing a comment preventing people with good intentions from reintroducing the stacking bug.
```suggestion
				/* NB: allies (cf. `teamHandler.Ally`) are also blocked
				 * even though in theory we could want to assist them. This
				 * is because allies aren't caught by `GetClosestFriendlyUnit`
				 * so it would be possible to build on top of them if they
				 * weren't assistable. */
				if (allyteam != cu->allyteam)
```